.dotnet/examples/Chat/Example04_FunctionCallingStreaming.cs

```diff
@@ -12,6 +15,122 @@ public partial class ChatExamples
 {
     // See Example03_FunctionCalling.cs for the tool and function definitions.
 
+    #region
+    public class StreamingChatToolCallsBuilder
+    {
+        private readonly Dictionary<int, string> _indexToToolCallId = [];
+        private readonly Dictionary<int, string> _indexToFunctionName = [];
+        private readonly Dictionary<int, SequenceBuilder<byte>> _indexToFunctionArguments = [];
+
+        public void Append(StreamingChatToolCallUpdate toolCallUpdate)
+        {
+            // Keep track of which tool call ID belongs to this update index.
+            if (toolCallUpdate.Id != null)
+            {
+                _indexToToolCallId[toolCallUpdate.Index] = toolCallUpdate.Id;
+            }
+
+            // Keep track of which function name belongs to this update index.
+            if (toolCallUpdate.FunctionName != null)
+            {
+                _indexToFunctionName[toolCallUpdate.Index] = toolCallUpdate.FunctionName;
+            }
+
+            // Keep track of which function arguments belong to this update index,
+            // and accumulate the arguments as new updates arrive.
+            if (toolCallUpdate.FunctionArgumentsUpdate != null && !toolCallUpdate.FunctionArgumentsUpdate.ToMemory().IsEmpty)
+            {
+                if (!_indexToFunctionArguments.TryGetValue(toolCallUpdate.Index, out SequenceBuilder<byte> argumentsBuilder))
+                {
+                    argumentsBuilder = new SequenceBuilder<byte>();
+                    _indexToFunctionArguments[toolCallUpdate.Index] = argumentsBuilder;
+                }
+
+                argumentsBuilder.Append(toolCallUpdate.FunctionArgumentsUpdate);
+            }
+        }
+
+        public IReadOnlyList<ChatToolCall> Build()
+        {
+            List<ChatToolCall> toolCalls = [];
+
+            foreach ((int index, string toolCallId) in _indexToToolCallId)
+            {
+                ChatToolCall toolCall = ChatToolCall.CreateFunctionToolCall(
+                    toolCallId: toolCallId,
+                    functionName: _indexToFunctionName[index],
+                    functionArguments: BinaryData.FromBytes(_indexToFunctionArguments[index].Build().ToArray()));
+
+                toolCalls.Add(toolCall);
+            }
+
+            return toolCalls;
+        }
+    }
+    #endregion
+
+    #region
+    public class SequenceBuilder<T>
```

Comment:
This is @KrzysztofCwalina's `SequenceBuilder`, taken from here: 
🔗 https://gist.github.com/KrzysztofCwalina/5f1671bfa82a975f0fda29116b1d9ce4